backend/app.js

```diff
@@ -33,19 +30,27 @@ const server = new ApolloServer({
       return { currentUser }
     }
   }
-});
-server.applyMiddleware({ app });
-
+})
+server.applyMiddleware({ app })
 
-const { MongoMemoryServer } = require('mongodb-memory-server')
 const mongoose = require('mongoose')
-const mongoServer = new MongoMemoryServer()
-mongoServer.getUri().then(uri => {
-  mongoose.connect(uri, { useNewUrlParser: true, useUnifiedTopology: true })
-  console.log('connected to mongodb')
-})
+if (process.env.NODE_ENV === 'development' || process.env.NODE_ENV === 'test') {
+  const { MongoMemoryServer } = require('mongodb-memory-server')
+  const mongoServer = new MongoMemoryServer()
+  mongoose.set('useFindAndModify', false)
+  mongoose.set('useCreateIndex', true)
+  mongoServer.getUri().then(uri => {
+    mongoose.connect(uri, { useNewUrlParser: true, useUnifiedTopology: true })
+    console.log('connected to mongodb')
+  })
+} else { // production mode
+  mongoose.set('useFindAndModify', false)
+  mongoose.set('useCreateIndex', true)
+  mongoose.connect('mongodb://luma-varaukset-db:27017/luma-varaukset', { useNewUrlParser: true, useUnifiedTopology: true })
+}
 
 const createAdmin = async () => {
+  await User.deleteMany({})
```

Comment:
Tämä siksi, että jos tietokanta on jo olemassa niin homma kaatuu muuten "Username must be unique"-virheeseen.